lib/OptimizationOptimisers/src/OptimizationOptimisers.jl

```diff
@@ -59,9 +59,19 @@ function SciMLBase.__solve(cache::OptimizationCache{
     else
         cache.solver_args.epochs
     end
+    maxiters = if cache.solver_args.maxiters === nothing
```

Comment:
This can be simplified since there's an overlap with the `nothing` and `nothing` condition above, you can assume that either `cache.solver_args.epochs` is not `nothing` or `cache.solver_args.maxiters` is not, will avoid repeating the error branch.